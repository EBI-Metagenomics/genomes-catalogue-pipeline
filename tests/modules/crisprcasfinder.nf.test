nextflow_process {

    name "Test Process CRISPRCasFinder"
    script "modules/crisprcasfinder.nf"
    process "CRISPRCAS_FINDER"

    test("Should run without failures") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = channel.from([
                    tuple(
                        "MGYG000000001",
                        file("$baseDir/tests/modules/fixtures/crisprcasfinder/MGYG000299389.fna")
                    )
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
