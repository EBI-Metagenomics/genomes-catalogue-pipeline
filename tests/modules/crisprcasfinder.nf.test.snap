{
    "Should run without failures": {
        "content": [
            {
                "hq_gff": [
                    [
                        "MGYG000000001",
                        "MGYG000299389.crisprcasfinder.hq.gff:md5,54842f3efcc24aeda4bc43947554929b"
                    ]
                ],
                "gff": [
                    [
                        "MGYG000000001",
                        "MGYG000299389.crisprcasfinder.gff:md5,dee757c84f1f24d6d7d307d2f619eba6"
                    ]
                ],
                "tsv": [
                    [
                        "MGYG000000001",
                        "MGYG000299389.crisprcasfinder.tsv:md5,2dbe0adb38bd359eca7eb980f4f6daf9"
                    ]
                ]
            }
        ],
        "timestamp": "2023-03-22T20:56:26+0000"
    }
}
